elasticapm/instrumentation/packages/asyncio/aiohttp_client.py

```diff
@@ -72,6 +72,14 @@ def mutate_unsampled_call_args(self, module, method, wrapped, instance, args, kw
         )
 
         headers = kwargs.get("headers", {})
-        headers[constants.TRACEPARENT_HEADER_NAME] = trace_parent.to_string()
+        self._set_disttracing_headers(headers, trace_parent, transaction)
         kwargs["headers"] = headers
         return args, kwargs
+
+    def _set_disttracing_headers(self, headers, trace_parent, transaction):
+        trace_parent_str = trace_parent.to_string()
+        headers[constants.TRACEPARENT_HEADER_NAME] = trace_parent_str
+        if transaction.tracer.config.use_elastic_traceparent_header:
+            headers[constants.TRACEPARENT_LEGACY_HEADER_NAME] = trace_parent_str
+        if trace_parent.tracestate:
+            headers[constants.TRACESTATE_HEADER_NAME] = trace_parent.tracestate
```

Comment:
We'll probably want to keep an eye on these types of functions as we go along. Has potential to become a maintenance problem if this behavior changes over time. Though I guess searching for the `constants.TRACEPARENT` will probably be sufficient for any number of uses.